tests/testsuite/lints/global_hint_mostly_unused.rs

```diff
@@ -0,0 +1,120 @@
+use crate::prelude::*;
+use cargo_test_support::project;
+use cargo_test_support::str;
+
+#[cargo_test(nightly, reason = "-Zhint-mostly-unused is unstable")]
+fn lint_global_mostly_unused() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+[package]
+name = "foo"
+version = "0.0.1"
+edition = "2015"
+
+[profile.dev]
+hint-mostly-unused = true
+"#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+    p.cargo("check -Zprofile-hint-mostly-unused -v")
+        .masquerade_as_nightly_cargo(&["profile-hint-mostly-unused", "cargo-lints"])
+        .with_stderr_data(str![[r#"
+[WARNING] `hint-mostly-unused` should not be applied globally
+ --> Cargo.toml:8:1
+  |
+7 | [profile.dev]
+8 | hint-mostly-unused = true
+  | ^^^^^^^^^^^^^^^^^^^^^^^^^
+  |
+  = [NOTE] `cargo::global_hint_mostly_unused` is set to `warn` by default
+[CHECKING] foo v0.0.1 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo [..]`
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
+        .run();
+}
+
+#[cargo_test(nightly, reason = "-Zhint-mostly-unused is unstable")]
+fn lint_global_mostly_unused_all_pkg_spec() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+[package]
+name = "foo"
+version = "0.0.1"
+edition = "2015"
+
+[profile.dev.package."*"]
+hint-mostly-unused = true
+"#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+    p.cargo("check -Zprofile-hint-mostly-unused -v")
+        .masquerade_as_nightly_cargo(&["profile-hint-mostly-unused", "cargo-lints"])
+        .with_stderr_data(str![[r#"
+[WARNING] `hint-mostly-unused` should not be applied globally
+ --> Cargo.toml:8:1
+  |
+7 | [profile.dev.package."*"]
+8 | hint-mostly-unused = true
+  | ^^^^^^^^^^^^^^^^^^^^^^^^^
+  |
+  = [NOTE] `cargo::global_hint_mostly_unused` is set to `warn` by default
```

Comment:
Not really a fan of the name `cargo::global_hint_mostly_unused`.  When it was originally mentioned out of context, I had no idea what it meant.  I suspect I'll still only know what it means when specific context is present.